File: DSA/Arshdeep-Kaur/Day 29/Day29_Question1.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int maximum = 0;
        int chunks = 0;

        for (int i=0; i<arr.size(); i++) {
            maximum = max(arr[i], maximum);
            if (i == maximum) {
                chunks++;
            }
        }

        return chunks;
    }
};

File: DSA/Pragya_Agarwal/Day29/Day29_Question1.py
class Solution:
    def maxChunksToSorted(self, arr: List[int]) -> int:
        n = len(arr)
        chunks = 0
        max_val = -1
    
        for i in range(n):
            max_val = max(max_val, arr[i])
            if i == max_val:
               chunks += 1
    
        return chunks
